minari/dataset/minari_storage.py

```diff
@@ -87,17 +88,53 @@ def apply(
 
         return out
 
+    def _h5_group_to_dict_recursive(self, hdf_ref, timestep):
+
+        if not isinstance(hdf_ref, h5py.Group):
+            return hdf_ref[timestep]
```

Comment:
unreacheable code?

Reply:
This is reachable since the type can be either `h5py.Dataset` or `h5py.Group`. I've made some changes that hopefully make this clearer to a reader.